sound/soc/sof/intel/hda-bus.c

```diff
@@ -20,10 +21,32 @@
 #endif
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+static void sof_hda_bus_link_power(struct hdac_device *codec, bool enable)
+{
+	struct hdac_bus *bus = codec->bus;
+	bool oldstate = test_bit(codec->addr, &bus->codec_powered);
+
+	snd_hdac_ext_bus_link_power(codec, enable);
```

Comment:
does this have to be called if the state isn't changing? It will check that internally presumably, but since you're checking here as well anyway...

Reply:
@lyakh Ack, this has to be called always as it updates to bus->codec_powered mask that is used in  many places.

Reply:
@kv2019i sorry, I mean
```
void snd_hdac_ext_bus_link_power(struct hdac_device *codec, bool enable)
{
	struct hdac_bus *bus = codec->bus;
	bool oldstate = test_bit(codec->addr, &bus->codec_powered);

	if (enable == oldstate)
		return;
```
if the state doesn't check, then that function is a NOP, so you could move your `if` from below to above

Reply:
@lyakh Oh yeah, you are right. In earlier implemented I just stored old state, call the generic link_power() and then compared whether anything has changed, but now that I changed to test_bit() directly, you are right, if() could be moved up.

The PR got merged upstream already, so I'll have to do this as a cosmetic fixup in a separate patch.